test/python/transpiler/test_clifford_t_passmanager.py

```diff
@@ -142,6 +142,26 @@ def test_qft(self, optimization_level):
         transpiled = pm.run(qc)
         self.assertLessEqual(set(transpiled.count_ops()), set(basis_gates))
 
+    @data(0, 1, 2, 3)
+    def test_multiplier(self, optimization_level):
+        """Clifford+T transpilation of a multiplier gate, using different optimization levels."""
+        gate = MultiplierGate(4)
+        qc = QuantumCircuit(gate.num_qubits)
+        qc.append(gate, qc.qubits)
+
+        # Transpile to a Clifford+T basis set
+        basis_gates = get_clifford_gate_names() + ["t", "tdg"]
+        pm = generate_preset_pass_manager(
+            basis_gates=basis_gates, optimization_level=optimization_level, seed_transpiler=0
+        )
+        transpiled = pm.run(qc)
+        self.assertLessEqual(set(transpiled.count_ops()), set(basis_gates))
+        t_count = _get_t_count(transpiled)
+
+        # expected T-count based on optimization level
+        expected_t_count = {0: 1085, 1: 1083, 2: 1071, 3: 1071}
+        self.assertEqual(_get_t_count(transpiled), 0)
```

Comment:
We don't use `t_count` or `expected_t_count` in the test, and the assertion is that there are _no_ T gates?  Should we also be counting `tdg` since it's in the basis set?

Hard-coding exact values is possibly going to make the test annoying to update each time we make an improvement to the preset pass managers: this sort of thing  might be better tracked by ASV benchmarks rather than a static test, perhaps?